cmake/modules/BuildDPDK.cmake

```diff
@@ -0,0 +1,101 @@
+function(do_build_dpdk)
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "i386")
+    set(arch "x86_64")
+    set(machine "native")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i686")
+    set(arch "i686")
+    set(machine "native")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64")
+    set(arch "x86_64")
+    set(machine "native")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm|ARM")
+    set(arch "arm")
+    set(machine "armv7a")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
+    set(arch "arm64")
+    set(machine "armv8a")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(powerpc|ppc)64")
+    set(arch "ppc_64")
+    set(machine "power8")
+  else()
+    message(FATAL "not able to build DPDK support: "
+      "unknown arch \"${CMAKE_SYSTEM_PROCESSOR}\"")
+  endif()
+
+  if(CMAKE_C_COMPILER_ID STREQUAL GNU)
+    set(toolchain "gcc")
+  elseif(CMAKE_C_COMPILER_ID STREQUAL Clang)
+    set(toolchain "clang")
+  elseif(CMAKE_C_COMPILER_ID STREQUAL Intel)
+    set(toolchain "icc")
+  else()
+    message(FATAL "not able to build DPDK support: "
+      "unknown compiler \"${CMAKE_C_COMPILER_ID}\"")
+  endif()
+
+  if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+    set(execenv "linuxapp")
+  elseif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+    set(execenv "bsdapp")
+  else()
+    message(FATAL "not able to build DPDK support: "
+      "unsupported OS \"${CMAKE_SYSTEM_NAME}\"")
+  endif()
+
+  set(target "${arch}-${machine}-${execenv}-${toolchain}")
+
+  execute_process(
+    COMMAND $(MAKE) showconfigs
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/spdk/dpdk
+    OUTPUT_VARIABLE supported_targets)
+
+  if(NOT supported_target MATCHES ${target})
+    message(FATAL "not able to build DPDK support: "
+      "unsupported target \"${target}\"")
+  endif()
+
+  if(WITH_KERNEL_DIR)
+    if(NOT IS_DIRECTORY ${WITH_KERNEL_DIR})
```

Comment:
@liupan1111 this is my biggest concern. the target machine's kernel is not always the same as the build host. even we managed to fulfill the build-requirement by installing the package of `linux-headers-\`uname -r\``. there is good chance that the built DPDK does not work for the deployed machine.

@liewegas what do you think?

Reply:
I'm not sure I understand the kernel version dependency.  Does the build have to be built against a specific kernel?  I.e., if we build on the centos 7.3 kernel will the built resuilt break on 7.4?  Or a build on the 16.04 kernel break on a later HWE kernel? 

As long as those scenerios behave okay, I think it's fine to continue.  If there is a hard kernel dependency, then I'm not sure what we should do.. that would suck.  :(

Reply:
For SLE, I'm pretty sure kernel module code needs to be integrated into, built as part of, and shipped in the kernel package itself. That may be an oversimplification, but certainly I wouldn't expect to be building kernel module packages via `ceph.spec`.

Reply:
@ddiss Can you vet my comment, above?

Reply:
yeah, to my best knowledge, kernel modules are not portable across different versions. anyway, i am disabling the kernel modules in DPDK. 

@liupan1111 could you help test the build w/o the kernel modules? thanks